types/node-hrx/index.d.ts

```diff
@@ -0,0 +1,50 @@
+// Type definitions for node-hrx 0.1
+// Project: https://github.com/issa-tseng/node-hrx#readme
+// Definitions by: Christophe Coevoet <https://github.com/stof>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.7
+
+/// <reference types="node" />
+
+import { Readable } from 'stream';
+
+export class Directory implements Iterable<string> {
+    constructor(path: string, contents?: Record<string, HrxItem>, comment?: string | null);
+    readonly path: string;
+    readonly contents: Record<string, HrxItem>;
+    readonly comment?: string;
+    get(path: string): HrxItem | undefined;
+    isDirectory(): this is Directory;
+    isFile(): this is File;
+    list(): string[];
+    [Symbol.iterator](): Iterator<string>;
+}
+
+export class Archive extends Directory {
+    constructor(contents?: Record<string, HrxItem>, comment?: string | null);
+    readonly path: '';
+}
+
+export class File {
+    constructor(path: string, body: string, comment?: string | null);
+    readonly path: string;
+    readonly body: string;
+    readonly comment?: string;
+    isDirectory(): this is Directory;
+    isFile(): this is File;
+}
+
+export type HrxItem = File | Directory;
+
+export class ParseProblem extends Error {
+    constructor(message: string, position: { line: number, col: number });
```

Comment:
this requires `details` props, doesn't?
https://github.com/issa-tseng/node-hrx/blob/c98c758139171e9bc810dc61f468b66463a21477/lib/util/problem.js#L5

Reply:
hmm, indeed. I missed it (not that the library seems to never populate them for ParseProblem, so I don't really know how to describe it

Reply:
no worries, we, or someone using this, can refine this later

Reply:
I've added type declared as opaque dictionary, with one know `path` property, should fit any scenario for now. Let me know if that is OK for you